### PR TITLE
Vulnerability Type And Vulnerability Subtype enhancements

### PR DESCRIPTION
**Pull request by 1411dolly0**

### CODE REVIEW

File: src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
+    COMMAND_INJECTION(77,null),
+
+    UNRESTRICTED_FILE_UPLOAD(434,null),
+
+    OPEN_REDIRECT_3XX_STATUS_CODE(601,38),
+
+    // XXE Vulnerability
+    XXE(611,43);
+
+    public Integer CWE_ID;
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java:45]: Can you please create them as Private variables and expose them as public getters?

File: src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+    public Integer WASC_ID;
+
+    VulnerabilityType(Integer CWE_ID, Integer WASC_ID) {
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java:48]: Add the visibility as Private.

**Review by preetkaran20 — COMMENTED** [review]
Overall it looks good to me. Some cosmetic changes.

File: src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
+    PERSISTENT_XSS(79,8),
+    DOM_BASED_XSS(79,8),
+
+    // JWT Related
+    CLIENT_SIDE_VULNERABLE_JWT(null,null),
+    SERVER_SIDE_VULNERABLE_JWT(null,null),
+    INSECURE_CONFIGURATION_JWT(null,null),
+
+    PATH_TRAVERSAL(22,33),
+    COMMAND_INJECTION(77,null),
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java:36]: OS Commanding | WASC-31

**Comment by preetkaran20** [comment]
@1411dolly0, you also need to run ./gradlew spotlessApply for checkstye errors.

File: src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java [hunk 4]
```diff
@@ -208,11 +203,6 @@ private void addFacadeResourceInformation(
                         messageBundle.getString(description, null));
                 List<VulnerabilityType> facadeVulnerabilityTypes =
                         new ArrayList<VulnerabilityType>();
-                for (org.sasanlabs.vulnerability.types.VulnerabilityType vulnerabilityType :
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: This was referring to the Vulnerability type but we wanted to add the Vulnerability Subtype here and also include both CWE and WASC.

Comment (preetkaran20) [reply src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: So along with this code, we need to add the CWE and WASC.

Comment (1411dolly0) [reply src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: Hi

VulnerabilityDefinition has a List of VulnerabilityType, which has value(String) and identifierType(String), So can you please explain where to include CWE and WASC?

Comment (preetkaran20) [reply src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: oh ok,

something like:
```java
facadeVulnerabilityTypes.add(
                            new VulnerabilityType("Custom", vulnerabilityType.name())
facadeVulnerabilityTypes.add(
                            new VulnerabilityType("CWE", vulnerabilityType.getCWE())
facadeVulnerabilityTypes.add(
                            new VulnerabilityType("WASC", vulnerabilityType.getWASC())
```

Comment (1411dolly0) [reply src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: Hi

3. Remove VulnerabilityType from https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/internal/utility/annotations/VulnerableAppRestController.java annotation

We removed VulnerabilityType[] from VulnerableAppRestController.java, on which we were iterating.

Not sure, how to proceed.

Comment (preetkaran20) [reply src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:211]: sorry for the confusion, so these changes should be done at https://github.com/SasanLabs/VulnerableApp/pull/328/files/4a4b5cf61c0313c19dbdd6f106a7b4a2c6ee4a1e#diff-2b30db6377c09a398f894a2b738e5d797896762ee30e6a5d605f08a8416fdcceL238 place.

You need to get the Vulnerability type from the attack vector annotation and then use it.

**Comment by 1411dolly0** [comment]
Pull request updated. Please review.

File: src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java [hunk 5]
```diff
@@ -235,10 +225,26 @@ private void addFacadeResourceInformation(
                         for (AttackVector attackVector : attackVectors) {
                             List<VulnerabilityType> facadeLevelVulnerabilityTypes =
                                     new ArrayList<VulnerabilityType>();
+                            org.sasanlabs.vulnerability.types.VulnerabilityType[]
+                                    vulnerabilityTypes = attackVector.vulnerabilityExposed();
                             for (org.sasanlabs.vulnerability.types.VulnerabilityType
                                     vulnerabilityType : vulnerabilityTypes) {
                                 facadeLevelVulnerabilityTypes.add(
                                         new VulnerabilityType("Custom", vulnerabilityType.name()));
+                                facadeVulnerabilityTypes.add(
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/service/impl/EndPointsInformationProvider.java:234]: don't add the CWE and WASC if they are having null value. Also can we avoid ternary operator?

File: src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java [hunk 6]
```diff
@@ -2,15 +2,46 @@
...
+    COMMAND_INJECTION(77, 31),
+
+    UNRESTRICTED_FILE_UPLOAD(434, null),
+
+    OPEN_REDIRECT_3XX_STATUS_CODE(601, 38),
+
+    // XXE Vulnerability
+    XXE(611, 43);
+
+    private Integer CWE_ID;
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/vulnerability/types/VulnerabilityType.java:32]: I suggest camel case, cweID and wascID and then generate the getter and setters as getCWEID(), getWASCID()

**Review by preetkaran20 — APPROVED** [review]
LGTM, thank you !!!!.

**Pull request merged**